src/main/java/uk/gov/pay/products/service/PaymentCreator.java

```diff
@@ -141,6 +142,11 @@ private PaymentEntity mergePaymentEntityWithReferenceNumberCheck(PaymentEntity p
                         kv(PAYMENT_EXTERNAL_ID, paymentEntity.getGovukPaymentId()),
                         kv("product_external_id", paymentEntity.getProductEntity().getExternalId())
                 );
+                
+                if (PanDetector.isSuspectedPan(paymentEntity.getReferenceNumber())) {
+                    logger.warn("Suspected PAN entered by user in reference field for payment id {}", 
+                            paymentResponse.getPaymentId());
```

Comment:
could this use structured logging?
`kv(PAYMENT_EXTERNAL_ID, externalId)`

Reply:
yep. can you tell I've been away from this for a while...